packages/config/src/layer2s/templates/opStack.ts

```diff
@@ -91,6 +96,14 @@ export function opStack(templateVars: OpStackConfig): Layer2 {
   const daProvider =
     templateVars.daProvider ??
     (postsToCelestia ? CELESTIA_DA_PROVIDER : undefined)
+
+  if (daProvider === undefined) {
+    assert(
```

Comment:
can it be done at the type level?

Reply:
No, because we are now fetching things from discovery. So only at run time you'll know for sure if a chain is an optimum or not.